fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/Wallet.java

```diff
@@ -0,0 +1,327 @@
+   *
+   * @param path a String path of the form: m/i/j/k/... where i,j,k,.. are 31-bit integer.
+   * @return a Pair<byte[], byte[]> representing the keys pair: first=private_key; second=public_key.
+   * @throws NoSuchAlgorithmException
+   * @throws InvalidKeyException
+   * @throws ShortBufferException
+   */
+  public Pair<byte[], byte[]> generateKeyFromPath(String path)
+      throws NoSuchAlgorithmException, InvalidKeyException, ShortBufferException {
+    if (path == null) {
+      throw new IllegalArgumentException("Unable to find keys from a null path!");
+    }
+    //split the path string
+    List<String> pathValue = new ArrayList<>(Arrays.asList(path.split("/")));
+    Log.d(TAG, "Path decomposed: " + pathValue);
+
+    pathValue.remove(0); //remove the first element (m)
+
+    //convert the path string in an array of int
+    int[] pathValueInt = pathValue
+        .stream()
+        .map(Integer::parseInt)
+        .mapToInt(Integer::intValue).toArray();
+
+    // derive private and public key
+    byte[] privateKey = deriveEd25519PrivateKey(seed, pathValueInt);
+    Ed25519PrivateKeyParameters prK = new Ed25519PrivateKeyParameters(privateKey, 0);
+    Ed25519PublicKeyParameters puK = prK.generatePublicKey();
+    byte[] publicKey = puK.getEncoded();
+
+    return new Pair<>(privateKey, publicKey);
+  }
+
+  /**
+   * Method that allows generate keys from the ID of the LAO and the ID of the RollCall.
+   *
+   * @param laoID a String.
+   * @param rollCallID a String.
+   * @return a Pair<byte[], byte[]> representing the keys pair.
+   * @throws NoSuchAlgorithmException
+   * @throws InvalidKeyException
+   * @throws ShortBufferException
+   */
+  public Pair<byte[], byte[]> findKeyPair(String laoID, String rollCallID)
+      throws NoSuchAlgorithmException, InvalidKeyException, ShortBufferException {
+    if (laoID == null || rollCallID == null) {
+      throw new IllegalArgumentException("Unable to find keys from a null param");
+    }
+    //Generate the string path
+    StringJoiner joiner = new StringJoiner("/");
+    joiner.add("m");
+    joiner.add(Integer.toString(PURPOSE));
+    joiner.add(Integer.toString(ACCOUNT));
+    joiner.add(convertStringToPath(laoID));
+    joiner.add(convertStringToPath(rollCallID));
+    String res = joiner.toString();
+
+    Log.d(TAG, "Generated path: " + res);
+
+    return generateKeyFromPath(res);
+  }
+
+  private String convertStringToPath(String string){
+    // extract byte form string
+    byte[] byteString = Base64.getDecoder().decode(string);
+
+    // create 31-bit index path
+    StringJoiner joiner = new StringJoiner("/");
+    int i = 0;
+    while(i < byteString.length){
+      int res = 0;
+      for(int j=i; j<i+3 && j< byteString.length; j++){
+        int buffer =  byteString[j];
+        buffer = buffer & 0xFF; // "extract" the first 8 bit
+        res = (buffer << j%3*8) | res;
+      }
+      i+= 3;
```

Comment:
Adding some comments explaining this implementation could be helpful

Reply:
Okay, I added a small comment on the top of the function.